src/modules/sub-items/components/bulk-delete/bulk.delete.btn.js

```diff
@@ -0,0 +1,163 @@
+import React, { PureComponent, Fragment } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+
+import Popup from '../../../common/popup/popup.component';
+import { bulkDeleteContents } from './../../services/bulk.service.js';
+
+import './css/bulk.delete.btn.css';
+
+class BulkDeleteButton extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.onBtnClick = this.onBtnClick.bind(this);
+        this.closePopup = this.closePopup.bind(this);
+        this.onPopupConfirm = this.onPopupConfirm.bind(this);
+
+        this.state = {
+            isPopupVisible: false,
+        };
+    }
+
+    componentDidMount() {
+        this.modalContainer = document.createElement('div');
+        this.modalContainer.classList.add('c-bulk-delete-btn__modal-container');
+        document.body.appendChild(this.modalContainer);
+    }
+
+    componentWillUnmount() {
+        document.body.removeChild(this.modalContainer);
+    }
+
+    onBtnClick() {
+        this.togglePopup(true);
+    }
+
+    bulkDelete() {
+        const { restInfo, selectedItems: itemsToDelete, removeItemsFromList } = this.props;
+        const contentsToDelete = itemsToDelete.map((item) => item.content);
+
+        bulkDeleteContents(restInfo, contentsToDelete, (deletedContents, notDeleted) => {
+            const deletedContentsIds = deletedContents.map((content) => content._id);
+
+            removeItemsFromList((item) => deletedContentsIds.includes(item.content._id));
+
+            if (deletedContents.length > 0) {
+                const message = Translator.trans(
+                    /*@Desc("The selected content item(s) have been sent to trash")*/ 'bulk_delete.success.message',
+                    {},
+                    'sub_items'
+                );
+
+                window.eZ.helpers.notification.showSuccessNotification(message);
```

Comment:
As far as I understand the service there might be a situation when both `deletedContents` and `notDeleted` have items. Displaying two notifications:
- `The selected content item(s) have been sent to trash`
- `You do not have permission to delete at least 1 of the selected content item(s). Please contact your Administrator to obtain permissions.`

Might be confusing for the users.

Reply:
Yes, I was planning to write about this to PMs.